core/src/main/java/org/apache/iceberg/BaseMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.Serializable;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.iceberg.util.PropertyUtil;
+import org.immutables.value.Value;
+
+/**
+ * A base class for {@link TableMetadata} and {@link org.apache.iceberg.view.ViewMetadata} It
+ * abstracts out all the common metadata for table ane view.
+ */
+public interface BaseMetadata extends Serializable {
```

Comment:
Just one question. Is this refactoring enough? 

If we Introduce `BaseMetadata`, I expect the `TableOperations` and `ViewOperations` to have a base class called `BaseOperations` which uses this `BaseMetadata` for interfaces like 
`void commit(BaseMetadata base, BaseMetadata metadata);`

Without that I don't see much value in this refactoring.

Reply:
may be we can have an end to end refactoring DRAFT PR and take a decision based on that.

Reply:
I agree, But i think those changes we can do later once this interface pattern gets approved. Or may be with this patch itself. 
WDYT ?

Reply:
we can have a separate draft patch. I think doing that refactoring can help us know whether the current interfaces in `BaseMetadata` is enough or do we need more changes here.

Reply:
Yea I think the main refactor is over at https://github.com/apache/iceberg/pull/8907 (including baseOperations and everything).  I do agree that it is a bit hard to read, I have to search over there to find what is needed and not, and so may miss something :).

Reply:
Actually sorry the refactor of the Operations is split in https://github.com/apache/iceberg/pull/9461, #8907 is the whole patch.

Reply:
This is draft [PR](https://github.com/nk1506/iceberg/pull/10/files) . with https://github.com/apache/iceberg/pull/9461 and https://github.com/apache/iceberg/pull/9682 .